packages/cockroach/extra/register.py

```diff
@@ -278,21 +278,69 @@ def _init_cockroachdb_cluster(
     """
     @retrying.retry(
         wait_fixed=1000,
-        stop_max_attempt_number=10,
+        stop_max_attempt_number=60,
         retry_on_result=lambda x: x is False,
-        )
-    def _wait_for_pidfile(pid: int) -> bool:
+    )
+    def _wait_for_cluster_init() -> bool:
         """
-        Try and read the PID from `PID_FILE_PATH` and match it to the given `pid`.
-        If, after 10 seconds, the PIDs don't match, raise an exception.
+        CockroachDB Cluster initialization takes a certain amount of time
+        while the cluster ID and and node ID are written to the storage.
+
+        We wait 1 second to avoid having a huge amount of logs in the journal.
+        We retry for up to 60 attempts because, waiting an extra minute is not
+        a big deal and we have never seen this take longer than that.
+
+        If after 60 attempts the cluster ID is not available raise an exception.
         """
+        gossip_url = 'http://localhost:8090/_status/gossip/local'
+        try:
+            response = requests.get(
+                gossip_url,
+                # We expect this request to get a response very quickly
+                # since it is a request to localhost.
```

Comment:
The network connectivity is not a strong argument here.

More important is the question whether the code, the HTTP handler, is expected to serve _quickly_ or if it may sometimes be slow (but operational).

Is we ever run into a situation where this request would be reproducibly and accurately served after e.g. 2 seconds then this tight timeout control would result in an unnecessary crash loop. Let's please explain why we know that this cannot happen, otherwise let's set this to something more conservative like 30 seconds where I think it's much more obvious that we want to stop waiting.